tensorflow/python/ops/array_ops.py

```diff
@@ -2721,11 +2721,13 @@ def batch_gather(params, indices, name=None):
       raise ValueError("batch_gather does not allow indices with unknown "
                        "shape.")
     batch_indices = indices
+    dtype = indices.dtype
     accum_dim_value = 1
     for dim in range(ndims-1, 0, -1):
       dim_value = params_shape[dim-1]
-      accum_dim_value *= params_shape[dim]
-      dim_indices = gen_math_ops._range(0, dim_value, 1)
+      accum_dim_value *= gen_math_ops.cast(params_shape[dim], dtype)
```

Comment:
Please add a unit type that depends on this change.

Reply:
Shall I create a new file array_ops_test.py?

Reply:
It already exists in tensorflow/python/kernel_tests/array_ops_test.py